### PR TITLE
`azurerm_sentinel_watchlist` - Add required property `item_search_key`

### LINKED ISSUE #15851: `properties` not properly set in `azurerm_sentinel_watchlist` (missing `itemsSearchKey`, `rawContent`)
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

 - Terraform: v1.1.6
 - AzureRM Provider: v2.99.0

### Affected Resource(s)

* `azurerm_sentinel_watchlist` ([ref](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/sentinel_watchlist))

### Terraform Configuration Files

Any configuration will work to reproduce, for example:

```hcl
resource "azurerm_sentinel_watchlist" "example" {
  name                       = "example-watchlist"
  log_analytics_workspace_id = azurerm_log_analytics_solution.example.workspace_resource_id
  display_name               = "example-wl"
}
```

### Debug Output

Not really required as I know what the problem is (see below) - here is a short error log:

```
│ Error: creating Watchlist: (Name "MyWatchlist" / Workspace Name "my-workspace" / Resource Group "my-resource-group"): securityinsight.WatchlistsClient#CreateOrUpdate: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="400" Message="There is an issue with the payload : Required property 'itemsSearchKey' not found in JSON. Path 'properties', line 7, position 3."
│
│   with module.sentinel_watchlist["MyWatchlist"].azurerm_sentinel_watchlist.watchlist,
│   on ../../../terraform/watchlists/main.tf line 7, in resource "azurerm_sentinel_watchlist" "watchlist":
│    7: resource "azurerm_sentinel_watchlist" "watchlist" {
│
│ creating Watchlist: (Name "MyWatchlist" / Workspace Name
│ "my-workspace" / Resource Group "my-resource-group"):
│ securityinsight.WatchlistsClient#CreateOrUpdate: Failure responding to
│ request: StatusCode=400 -- Original Error: autorest/azure: Service returned
│ an error. Status=400 Code="400" Message="There is an issue with the payload
│ : Required property 'itemsSearchKey' not found in JSON. Path 'properties',
│ line 7, position 3."
```

### Expected Behaviour

When creating a new watchlist, it should not result in an error.

### Actual Behaviour

It errors.

### Steps to Reproduce

Pick any valid tf config to deploy `azurerm_sentinel_watchlist`

1. `terraform apply`
2. error will be produced

### Important Factoids

The problem is that the required  property `properties.itemsSearchKey` ([ref](https://docs.microsoft.com/en-us/rest/api/securityinsights/stable/watchlists/create-or-update)) are not being sent.

While not documented, to actually set such a search key, we also need to set `rawContent` otherwise we get:

```json
{
  "error": {
    "code": "400",
    "message": "There is an issue with the payload: the payload rawContent is empty!"
  }
}
```
To fix this problem: The resource will need to add two new arguments `items_search_key` and `raw_content`.

### References

- [ New resource: `azurerm_sentinel_watchlist`](https://github.com/hashicorp/terraform-provider-azurerm/pull/14258)

### PR DESCRIPTION
**Pull request by magodo**
Since March, the sentinel API now requires the `itemSearchey` in the `PUT` request. Meanwhile, the `contentType` is now not needed here as we don't actually send any content, but delegate it to the `azurerm_sentinel_watchlist_item`.

Fixes #15851.
Superseds: #15856.

## Test

```shell
💤  TF_ACC=1 go test -v -timeout=20h ./internal/services/sentinel  -run='TestAccWatchlist'
=== RUN   TestAccWatchlistItem_basic
=== PAUSE TestAccWatchlistItem_basic
=== RUN   TestAccWatchlistItem_update
=== PAUSE TestAccWatchlistItem_update
=== RUN   TestAccWatchlistItem_requiresImport
=== PAUSE TestAccWatchlistItem_requiresImport
=== RUN   TestAccWatchlist_basic
=== PAUSE TestAccWatchlist_basic
=== RUN   TestAccWatchlist_complete
=== PAUSE TestAccWatchlist_complete
=== RUN   TestAccWatchlist_requiresImport
=== PAUSE TestAccWatchlist_requiresImport
=== CONT  TestAccWatchlistItem_basic
=== CONT  TestAccWatchlist_requiresImport
=== CONT  TestAccWatchlist_basic
=== CONT  TestAccWatchlistItem_requiresImport
=== CONT  TestAccWatchlist_complete
=== CONT  TestAccWatchlistItem_update
--- PASS: TestAccWatchlist_basic (124.25s)
--- PASS: TestAccWatchlistItem_requiresImport (150.89s)
--- PASS: TestAccWatchlistItem_basic (184.24s)
--- PASS: TestAccWatchlistItem_update (228.30s)
--- PASS: TestAccWatchlist_requiresImport (784.88s)
--- PASS: TestAccWatchlist_complete (820.07s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/sentinel      820.092s
```

### CODE REVIEW

**Comment by magodo** [comment]
The CI failure of the example is due to there is new property introduced.

**Comment by mcamenzind** [comment]
Any reason for deviating from Microsoft's property naming `itemsSearchKey` (notice the `s`)?

File: internal/services/sentinel/sentinel_watchlist_resource.go [hunk 1]
```diff
@@ -124,10 +131,11 @@ func (r WatchlistResource) Create() sdk.ResourceFunc {
 					// The only supported provider for now is "Microsoft"
 					Provider: utils.String("Microsoft"),
 
-					// The "source" and "contentType" represent the source file name which contains the watchlist items and its content type.
+					// The "source" represent the source file name which contains the watchlist items.
 					// Setting them here is merely to make the API happy.
-					Source:      securityinsight.Source("a.csv"),
-					ContentType: utils.String("Text/Csv"),
```
Comment (tombuildsstuff) [inline internal/services/sentinel/sentinel_watchlist_resource.go:130]: this is a breaking change, we should be exposing a field for this with a default value here?

File: internal/services/sentinel/sentinel_watchlist_resource.go [hunk 2]
```diff
@@ -124,10 +131,11 @@ func (r WatchlistResource) Create() sdk.ResourceFunc {
...
+					Source: securityinsight.Source("a.csv"),
```
Comment (katbyte) [inline internal/services/sentinel/sentinel_watchlist_resource.go:136]: "a.csv" is not a valid value for securityinsight.Source? at that should we not expose this in schema with a default?

File: internal/services/sentinel/sentinel_watchlist_resource.go (hunk 1 above)
Comment (katbyte) [inline internal/services/sentinel/sentinel_watchlist_resource.go:130]: how come we are removing this? should it not be in schema with a default?

File: internal/services/sentinel/sentinel_watchlist_resource.go [hunk 3]
```diff
@@ -124,10 +131,11 @@ func (r WatchlistResource) Create() sdk.ResourceFunc {
...
+
+					ItemsSearchKey: utils.String(model.ItemsSearchKey),
```
Comment (katbyte) [inline internal/services/sentinel/sentinel_watchlist_resource.go:138]: ditto here? should this not just be in schema with a default?

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @magodo - left some comments inline. as for the the name `items_seach_key` sounds incorrect? correct grammar should be `item_search_key` regardless of what the SDK is using?

**Comment by magodo** [comment]
@tombuildsstuff and @katbyte The `Source` and `ContentType` are not returned from API, they are used by the service during the `PUT` so that the service can create watchlist items based on the specified source. However, as is stated in #14258 this resource is only to create an empty watchlist, if users want to create watchlist items, they can then use `azurerm_sentinel_watchlist_item` resource.

Therefore, this PR removes the `ContentType` (as is actually optional in turns of API), and the value of the `Source` is actually doesn't have any real effect, but only to make API happy, which even can be regarded as not a breaking change for TF users.

**Comment by Andyandpandy** [comment]
Any progress here? 
This PR is blocking me from going > 3.0.x

**Review by katbyte — APPROVED** [review]
LGTM 🚜

**Pull request merged**